addons/stock/stock.py

```diff
@@ -484,10 +484,10 @@ def quants_get(self, cr, uid, location, product, qty, domain=None, restrict_lot_
 
     def apply_removal_strategy(self, cr, uid, location, product, quantity, domain, removal_strategy, context=None):
         if removal_strategy == 'fifo':
-            order = 'in_date, id'
+            order = 'in_date, location_id, package_id, lot_id, id'
             return self._quants_get_order(cr, uid, location, product, quantity, domain, order, context=context)
         elif removal_strategy == 'lifo':
-            order = 'in_date desc, id desc'
+            order = 'in_date desc, location_id desc, package_id desc, lot_id desc, id desc'
```

Comment:
Why do you put "desc" after "location_id, package_id and lot_id" here ?
For me, it only makes sense for "location_id". I would prefer to remove "desc", because, when you read the code, you think that it has been set for a good reason, which is not the case here.